src/realm/data_type.hpp

```diff
@@ -19,8 +19,21 @@
 #ifndef REALM_DATA_TYPE_HPP
 #define REALM_DATA_TYPE_HPP
 
+#include "realm/string_data.hpp"
+#include "realm/binary_data.hpp"
+#include "realm/olddatetime.hpp"
+
```

Comment:
This file previously had 0 `#include`s, which seems to be a nice property of such a seemingly fundamental file. Is this commit necessary for the purpose of the PR in the first place?